### PR TITLE
Send/Gift sticker flow 

### PR DESCRIPTION
**Pull request by mdhanif-simformsolutions**
### 🎯 Goal
-  Gift sticker in group and one-one chat

### CODE REVIEW

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/EmojiPickerViewController.swift [hunk 1]
```diff
@@ -81,7 +131,8 @@ extension EmojiPickerViewController: UITableViewDelegate, UITableViewDataSource
         guard let cell = tableView.dequeueReusableCell(withIdentifier: "PickerTableViewCell") as? PickerTableViewCell else {
             return UITableViewCell()
         }
-        cell.configure(with: packages[indexPath.row], downloadedPackage: downloadedPackage)
+        cell.btnDownload.addTarget(self, action: #selector(btnDownloadAction(_:)), for: .touchUpInside)
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/EmojiPickerViewController.swift:134]: @mdhanif-simformsolutions  Pls don't use outlets directly in cellForRow. Use closure or protocol instead.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/EmojiPickerViewController.swift [hunk 2]
```diff
@@ -93,13 +144,10 @@ extension EmojiPickerViewController: UITableViewDelegate, UITableViewDataSource
...
-        downloadedPackage.append(packageId)
-        tableView.reloadRows(at: [indexPath], with: .automatic)
-        if packages[indexPath.row].isDownload != "Y" {
-            StickerApiClient.downloadStickers(packageId: packages[indexPath.row].packageID ?? 0) { _ in }
-        } else {
-            StickerApiClient.hideStickers(packageId: packages[indexPath.row].packageID ?? 0, nil)
+        if let sendEmojiVc: SendEmojiViewController = SendEmojiViewController.instantiateController(storyboard: .wallet) {
+            sendEmojiVc.packageInfo = packages[indexPath.row]
+            sendEmojiVc.chatChannelController = chatChannelController
+            self.present(sendEmojiVc, animated: true)
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/EmojiPickerViewController.swift:150]: @mdhanif-simformsolutions  `self` not required here.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import UIKit
+import StreamChat
+import Nuke
+
+class GroupListPickerViewController: UIViewController {
+
+    //MARK: Outlets
+    @IBOutlet weak var tblMember: UITableView! {
+        didSet {
+            tblMember.register(TableViewCellChatUser.nib, forCellReuseIdentifier: TableViewCellChatUser.reuseId)
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift:17]: @mdhanif-simformsolutions Pls move register cell in viewDidLoad.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    var didSelectMember: ((ChatChannelMember) -> Void)?
+    private var members = [ChatChannelMember]()
+    private var currentPage : Int = 0
+    private var totalCount: Int = 0
+    private var isLoadingList : Bool = false
+    private var channelMember: ChatChannelMemberListController?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        guard let cid = controller?.cid else { return }
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift:32]: @mdhanif-simformsolutions  Pls create setupUI() function and move these code there.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+//  StreamChatUI
+//
+//  Created by Parth Kshatriya on 08/04/22.
+//
+
+import UIKit
+import StreamChat
+import Nuke
+
+class GroupListPickerViewController: UIViewController {
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift:12]: @mdhanif-simformsolutions  Pls remove these controller if not required.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/SendEmojiViewController.swift [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    @IBAction func btnCloseAction(_ sender: Any) {
+        self.dismiss(animated: true, completion: nil)
+    }
+}
+
+// MARK: - Collection view delegate
+@available(iOS 13.0, *)
+extension SendEmojiViewController: UICollectionViewDelegate { }
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/SendEmojiViewController.swift:108]: @mdhanif-simformsolutions  Pls remove UICollectionViewDelegate

**Review by ajay-simform — CHANGES_REQUESTED** [review]

File: Sources/StreamChat/APIClient/Endpoints/StickerEndPoints.swift [hunk 7]
```diff
@@ -11,13 +11,16 @@ import Combine
 @available(iOS 13.0, *)
 extension StickerApi {
     public enum RequestType: EndPointType {
-        case mySticker
+        case mySticker, getHiddenStickers
```
[the remark this answers is not in the document]
Comment (mdhanif-simformsolutions) [reply Sources/StreamChat/APIClient/Endpoints/StickerEndPoints.swift]: Updated this thanks.

File: Sources/StreamChat/APIClient/StickerApiClient.swift [hunk 8]
```diff
@@ -54,46 +60,102 @@ public struct Agent {
...
+        _ completion: @escaping ((ResponseBody<RecentStickerBody>) -> Void)
+    ) {
         StickerApi.call(type: .recentSticker)
             .sink { _  in } receiveValue: { result in completion(result) }
             .store(in: &stickerCalls)
     }
 
-    public static func hideStickers(packageId: Int, _ completion: ((ResponseBody<EmptyStipopResponse>) -> Void)?) {
+    public static func hideStickers(
+        packageId: Int, _ completion: ((ResponseBody<EmptyStipopResponse>) -> Void)?
```
[the remark this answers is not in the document]
Comment (mdhanif-simformsolutions) [reply Sources/StreamChat/APIClient/StickerApiClient.swift]: Updated this thanks.

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/StickerGiftTableViewCell.swift [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+    public private(set) var subContainer: UIView!
+    public private(set) var imgStickerPreview: UIImageView!
+    public private(set) var timestampLabel: UILabel!
+    public private(set) var descriptionLabel: UILabel!
+    public private(set) var pickUpButton: UIButton!
+    public private(set) var lblDetails: UILabel!
+    private var detailsStack: UIStackView!
+    var content: ChatMessage?
+    var isSender = false
+    var channel: ChatChannel?
```
[the remark this answers is not in the document]
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/StickerGiftTableViewCell.swift:25]: We are using this now to check if it's group or one-one chat.

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/StickerGiftTableViewCell.swift [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    // MARK: - Custom functions
+    func configureCell(isSender: Bool) {
+        self.isSender = isSender
+        viewContainer = UIView()
+        viewContainer.translatesAutoresizingMaskIntoConstraints = false
+        viewContainer.backgroundColor = .clear
+        viewContainer.clipsToBounds = true
+        contentView.addSubview(viewContainer)
```
[the remark this answers is not in the document]
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/StickerGiftTableViewCell.swift:52]: Updated this thanks.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/Cell/PickerTableViewCell.swift [hunk 11]
```diff
@@ -9,22 +9,34 @@ import Foundation
...
 import Nuke
 
+@available(iOS 13.0, *)
 class PickerTableViewCell: UITableViewCell {
     //MARK: Outlets
     @IBOutlet private weak var lblPackName: UILabel!
     @IBOutlet private weak var lblArtistName: UILabel!
     @IBOutlet private weak var imgPack: UIImageView!
-    @IBOutlet private weak var btnDownload: UIButton!
+    @IBOutlet weak var btnDownload: UIButton!
```
[the remark this answers is not in the document]
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/Cell/PickerTableViewCell.swift]: Updated this thanks.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/EmojiPickerViewController.swift (hunk 1 above)
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/EmojiPickerViewController.swift]: Updated this thanks.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/EmojiPickerViewController.swift (hunk 2 above)
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/EmojiPickerViewController.swift]: Updated this thanks.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift (hunk 3 above)
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift]: Removed this controller as it is not required for now.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift (hunk 4 above)
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift]: Removed this controller as it is not required for now.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/SendEmojiViewController.swift (hunk 6 above)
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/SendEmojiViewController.swift]: Updated this thanks.

File: Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift (hunk 5 above)
Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GroupListPickerViewController.swift]: Removed this controller as it is not required for now.

**Pull request merged**